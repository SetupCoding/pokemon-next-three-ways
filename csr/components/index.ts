export { Loading } from "./Loading";
export { ErrorComponent } from "./ErrorComponent";
export { PokemonCard } from "./PokemonCard";
export { Home } from "./Home";
export { LoadMore } from "./LoadMore";
