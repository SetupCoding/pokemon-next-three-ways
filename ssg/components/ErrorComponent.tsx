import { HttpError } from "../models/http-error.model";
import React from "react";
import { Row } from "react-bootstrap";
type Props = {
  error?: HttpError;
};
export const ErrorComponent: React.FC<Props> = ({ error }) =>
  error ? (
    <>
      <Row className="justify-content-md-center">{error.message}</Row>
    </>
  ) : null;
