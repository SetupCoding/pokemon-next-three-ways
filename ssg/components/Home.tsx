import Link from "next/link";
import React from "react";
export const Home: React.FC = () => (
  <Link href="/">
    <a>{"< Home"}</a>
  </Link>
);
