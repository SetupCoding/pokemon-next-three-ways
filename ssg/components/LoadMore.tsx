import React from "react";
export const LoadMore = ({ setPage }) => {
  const onLoadMore = () => {
    setPage((prevStart) => prevStart + 1);
  };
  return <button onClick={onLoadMore}>Load more</button>;
};
