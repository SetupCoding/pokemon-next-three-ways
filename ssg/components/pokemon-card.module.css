.cardWrapper {
    padding: 5px;
}

.card {
    cursor: pointer;
}

.cardImage{
    height: 300px;
    object-fit: contain;
}

.detailImage {
    height: 300px;
}

.cardWrapper> a{
    color: black;
    text-decoration: none;
}

.cardWrapper > a:hover {
    color: black;
    text-decoration: none;
}
