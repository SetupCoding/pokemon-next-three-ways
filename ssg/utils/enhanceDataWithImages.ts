import { HOST_URL } from "../constants/const";
import { Pokemon } from "../models/pokemon.model";

export const addImage = (pokemon: Pokemon) => ({
  ...pokemon,
  image: `/pokemon/${pokemon.name.english.toLowerCase().replace(" ", "-")}.jpg`,
});

export const enhanceDataWithImages = (
  data: Omit<Pokemon, "image">[]
): Pokemon[] => data.map((pokemon) => addImage(pokemon));
