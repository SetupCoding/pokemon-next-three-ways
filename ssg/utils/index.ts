export {enhanceDataWithImages} from './enhanceDataWithImages';
export {handleErrors} from './handleErrors';