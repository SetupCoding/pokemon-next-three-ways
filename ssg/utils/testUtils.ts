import React from "react";
import renderer from "react-test-renderer";

export const testSnapshot = (component) => {
  return expect(renderer.create(component).toJSON()).toMatchSnapshot();
};
