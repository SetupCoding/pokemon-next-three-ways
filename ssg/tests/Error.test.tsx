import { ErrorComponent } from "../components";
import React from "react";
import { testSnapshot } from "../utils/testUtils";

describe("<ErrorComponent />", () => {
  it("renders the ErrorComponent component correctly", () => {
    testSnapshot(
      <ErrorComponent
        error={{ status: 404, error: "Not Found", message: "Not Found" }}
      />
    );
  });
});
