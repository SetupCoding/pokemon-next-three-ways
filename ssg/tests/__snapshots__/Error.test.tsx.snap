// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ErrorComponent /> renders the ErrorComponent component correctly 1`] = `
<div
  className="justify-content-md-center row"
>
  Not Found
</div>
`;
